ShapeAnalysis/python/PlotFactory.py

```diff
@@ -458,7 +458,17 @@ def makePlot(self, inputFile, outputDirPlots, variables, cuts, samples, plot, nu
                       else:
                         histoVar = fileIn.Get(shapeNameVar)
   
-                      nuisanceHistos[ivar][nuisanceName] = histoVar
+                      if histoVar != None :
+                        nuisanceHistos[ivar][nuisanceName] = histoVar
+                      elif not self._SkipMissingNuisances :
+                        print " This is bad, the nuisance ", nuisanceName, " is missing! You need to add it, maybe some jobs crashed?"
+                        nuisanceHistos[ivar][nuisanceName] = histoVar
+                      else :
+                        # if you had self._SkipMissingNuisances set to true, put the variation the same as the nominal
+                        histoVar = histo.Clone(shapeNameVar.replace('/', '__'))
+                        histoVar.Scale(variations[ivar])
```

Comment:
variations is not defined here for shape uncertainties

Reply:
ok, no scaling is needed, thanks!